### PR TITLE
Normalize the Balancing Authority Table and add a BA Association Table

### PR DESCRIPTION
**Pull request by zaneselvans**
Really just one big function in here, that makes use of a bunch of the transformed dataframes to compile an association table for the balancing authorities, and also it makes the structure of the balancing authority table uniform across the years.

### CODE REVIEW

File: src/pudl/transform/eia861.py [hunk 1]
```diff
@@ -1091,8 +1239,8 @@ def transform(raw_dfs, eia861_tables=pc.pudl_tables["eia861"]):
     """
     # these are the tables that we have transform functions for...
     tfr_funcs = {
-        "service_territory_eia861": service_territory,
         "balancing_authority_eia861": balancing_authority,
+        "service_territory_eia861": service_territory,
         "sales_eia861": sales,
         "advanced_metering_infrastructure_eia861": advanced_metering_infrastructure,
         "demand_response_eia861": demand_response,
```
Comment (aesharpe) [inline src/pudl/transform/eia861.py:1246]: Dynamic pricing should probably be in this?

Comment (zaneselvans) [reply src/pudl/transform/eia861.py:1246]: Oh yeah definitely. Did I remove it somehow?

Comment (zaneselvans) [reply src/pudl/transform/eia861.py:1246]: Oh it is there -- it's just outside of the excerpt being shown here. You can expand the excerpts with the little arrows at the top and bottom of them.

Comment (aesharpe) [reply src/pudl/transform/eia861.py:1246]: Oh I see! Looks good then

**Pull request merged**